src/test/java/org/sciborgs1155/robot/HopperTest.java

```diff
@@ -0,0 +1,32 @@
+package org.sciborgs1155.robot;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.sciborgs1155.lib.TestingUtil.*;
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.sciborgs1155.robot.Hopper.Hopper;
+
+public class HopperTest {
+  Hopper hopper = new Hopper();
+  final double DELTA = 2;
+
+  @BeforeEach
+  public void setup() {
+    setupHAL();
```

Comment:
you should be making the hopper here. so at the top, you can just have
Hopper hopper;
and then in setup, you say hopper = new Hopper();